scripts/pull_image_descriptions.py

```diff
@@ -0,0 +1,542 @@
+#!/usr/bin/env python3
+
+import argparse
+import fnmatch
+import json
+import pathlib
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+from datetime import datetime
+from typing import Dict, List, Tuple
+
+CONTAINER_IMAGE = "ghcr.io/osbuild/image-builder-cli:latest"
+GENERATION_DATE = datetime.now().strftime("%Y-%m-%d %H:%M:%S UTC")
+TARGET_DIR = pathlib.Path(__file__).parent.parent / "docs" / "user-guide" / "09-image-descriptions"
+
+
+def run_command(cmd: List[str]) -> Tuple[bool, str, str]:
+    """
+    Run a command and return success status, stdout, and stderr.
+    """
+    try:
+        result = subprocess.run(cmd, capture_output=True, text=True, check=True)
+        return True, result.stdout, result.stderr
+    except subprocess.CalledProcessError as e:
+        return False, e.stdout, e.stderr
+
+
+def pull_container_image() -> bool:
+    """
+    Pull the latest image-builder-cli container image.
+    """
+    print("Pulling latest container image...")
+    pull_cmd = ["sudo", "podman", "pull", CONTAINER_IMAGE]
+    success, _, stderr = run_command(pull_cmd)
+    if not success:
+        print(f"Warning: Failed to pull container image: {stderr}")
+        return False
+    print("Container image pulled successfully")
+    return True
+
+
+def get_container_version() -> str:
+    """
+    Get the version/ref of the image-builder-cli container.
+    """
+    # Get the image ID and digest information
+    cmd = ["sudo", "podman", "images", CONTAINER_IMAGE, "--format", "{{.Repository}}:{{.Tag}}@{{.Digest}}"]
+    success, stdout, _ = run_command(cmd)
+    if success and stdout.strip():
+        return stdout.strip()
+
+    # Fallback to just the image ID if digest is not available
+    fallback_cmd = ["sudo", "podman", "images", CONTAINER_IMAGE, "--format", "{{.Repository}}:{{.Tag}} ({{.ID}})"]
+    success, stdout, _ = run_command(fallback_cmd)
+    if success and stdout.strip():
+        return stdout.strip()
+
+    return f"{CONTAINER_IMAGE} (unknown)"
+
+
+def list_images() -> Dict:
+    """
+    Get list of all supported images using list-images command.
+
+    The returned dict structure is
+    {
+        "distro_name": {
+            "arch_name": [image_type_name, image_type_name, ...],
+            "arch_name": [image_type_name, image_type_name, ...],
+        }
+    }
+    """
+    cmd = ["sudo", "podman", "run", "--rm", "--privileged", CONTAINER_IMAGE, "list-images", "--format", "json"]
```

Comment:
(nitpick) after some discussion "list-images" became just "list" (c.f.  "274a75387bdc34073506bd1d89e3b3245abfcff8" in image-builder-cli)

(idle-coment) you *could* just run `go run github.com/osbuild/image-builder-cli/cmd/image-builder@latest list --format=json` for listing the images (but this will not give you the git revision of ibcli so probably not useful here.